pkg/query-service/app/logs/parser.go

```diff
@@ -0,0 +1,306 @@
+	if len(strings.TrimSpace(tokenRegex.ReplaceAllString(query, ""))) > 0 {
+		return nil, fmt.Errorf("failed to parse query, contains unknown tokens")
+	}
+
+	for _, v := range filterTokens {
+		op := strings.TrimSpace(operatorRegex.FindString(v))
+		opLower := strings.ToLower(op)
+
+		if opLower == "contains" || opLower == "ncontains" {
+			searchString := strings.TrimSpace(strings.Split(v, op)[1])
+
+			operatorRemovedTokens := strings.Split(operatorRegex.ReplaceAllString(v, " "), " ")
+			searchCol := strings.ToLower(operatorRemovedTokens[0])
+			if searchCol == AND || searchCol == OR {
+				searchCol = strings.ToLower(operatorRemovedTokens[1])
+			}
+			col := searchCol
+			if strings.ToLower(searchCol) == "fulltext" {
+				col = "body"
+			}
+
+			f := fmt.Sprintf(`%s %s '%%%s%%' `, col, operatorMapping[opLower], searchString[1:len(searchString)-1])
+			if strings.HasPrefix(strings.ToLower(v), AND) {
+				f = "AND " + f
+			} else if strings.HasPrefix(strings.ToLower(v), OR) {
+				f = "OR " + f
+			}
+			sqlQueryTokens = append(sqlQueryTokens, f)
+		} else {
+			symbol := operatorMapping[strings.ToLower(op)]
+			sqlQueryTokens = append(sqlQueryTokens, strings.Replace(v, " "+op+" ", " "+symbol+" ", 1)+" ")
+		}
+	}
+
+	return sqlQueryTokens, nil
+}
+
+func parseColumn(s string) (*string, error) {
+	s = strings.ToLower(s)
+
+	colName := ""
+
+	// if has and/or as prefix
+	filter := strings.Split(s, " ")
+	if len(filter) < 3 {
+		return nil, fmt.Errorf("incorrect filter")
+	}
+
+	if strings.HasPrefix(s, AND) || strings.HasPrefix(s, OR) {
+		colName = filter[1]
+	} else {
+		colName = filter[0]
+	}
+
+	return &colName, nil
+}
+
+func arrayToMap(fields []model.LogField) map[string]model.LogField {
+	res := map[string]model.LogField{}
+	for _, field := range fields {
+		res[field.Name] = field
+	}
+	return res
+}
+
+func replaceInterestingFields(allFields *model.GetFieldsResponse, queryTokens []string) ([]string, error) {
+	// check if cols
+	selectedFieldsLookup := arrayToMap(allFields.Selected)
+	interestingFieldLookup := arrayToMap(allFields.Interesting)
+
+	for index := 0; index < len(queryTokens); index++ {
+		queryToken := queryTokens[index]
+		col, err := parseColumn(queryToken)
+		if err != nil {
+			return nil, err
+		}
+
+		sqlColName := *col
+		if _, ok := selectedFieldsLookup[*col]; !ok && *col != "body" {
+			if field, ok := interestingFieldLookup[*col]; ok {
+				sqlColName = fmt.Sprintf("%s_%s_value[indexOf(%s_%s_key, '%s')]", field.Type, strings.ToLower(field.DataType), field.Type, strings.ToLower(field.DataType), *col)
+			} else if strings.Compare(strings.ToLower(*col), "fulltext") != 0 {
+				return nil, fmt.Errorf("field not found for filtering")
+			}
+		}
+		queryTokens[index] = strings.Replace(queryToken, *col, sqlColName, 1)
+	}
+	return queryTokens, nil
+}
+
+func GenerateSQLWhere(allFields *model.GetFieldsResponse, params *model.LogsFilterParams) (string, error) {
```

Comment:
@nityanandagohain Should this function be part of parser.go?
I think all sql related functions should be moved to sql.go

Reply:
interesting, creating a new issue to address this.

Reply:
https://github.com/SigNoz/signoz/issues/1495